core/src/env/call/call_builder.rs

```diff
@@ -0,0 +1,447 @@
+
+mod seal {
+    /// Used to prevent users from implementing `IndicateReturnType` for their own types.
+    pub trait Sealed {}
+    impl Sealed for () {}
+    impl<T> Sealed for super::ReturnType<T> {}
+}
+/// Types that can be used in [`CallBuilder::returns`] to signal return type.
+pub trait IndicateReturnType: Default + self::seal::Sealed {}
+impl IndicateReturnType for () {}
+impl<T> IndicateReturnType for ReturnType<T> {}
+
+impl<E, Callee, GasLimit, TransferredValue, Args>
+    CallBuilder<E, Callee, GasLimit, TransferredValue, Args, Unset<ReturnType<()>>>
+where
+    E: EnvTypes,
+{
+    /// Sets the value transferred upon the execution of the call.
+    ///
+    /// # Note
+    ///
+    /// Either use `.returns::<()>` to signal that the call does not return a value
+    /// or use `.returns::<ReturnType<T>>` to signal that the call returns a value of
+    /// type `T`.
+    #[inline]
+    pub fn returns<R>(
+        self,
+    ) -> CallBuilder<E, Callee, GasLimit, TransferredValue, Args, Set<R>>
+    where
+        R: IndicateReturnType,
+    {
+        CallBuilder {
+            env_types: Default::default(),
+            callee: self.callee,
+            gas_limit: self.gas_limit,
+            transferred_value: self.transferred_value,
+            exec_input: self.exec_input,
+            return_type: Set(Default::default()),
+        }
+    }
+}
+
+impl<E, Callee, GasLimit, TransferredValue, RetType>
+    CallBuilder<
+        E,
+        Callee,
+        GasLimit,
+        TransferredValue,
+        Unset<ExecutionInput<EmptyArgumentList>>,
+        RetType,
+    >
+where
+    E: EnvTypes,
+{
+    /// Sets the execution input to the given value.
+    pub fn exec_input<Args>(
+        self,
+        exec_input: ExecutionInput<Args>,
+    ) -> CallBuilder<
+        E,
+        Callee,
+        GasLimit,
+        TransferredValue,
+        Set<ExecutionInput<Args>>,
+        RetType,
+    > {
+        CallBuilder {
+            env_types: Default::default(),
+            callee: self.callee,
+            gas_limit: self.gas_limit,
+            transferred_value: self.transferred_value,
+            exec_input: Set(exec_input),
+            return_type: self.return_type,
+        }
+    }
+}
+
+impl<E, GasLimit, TransferredValue, Args, RetType>
+    CallBuilder<
+        E,
+        Set<E::AccountId>,
+        GasLimit,
+        TransferredValue,
+        Set<ExecutionInput<Args>>,
+        Set<RetType>,
+    >
+where
+    E: EnvTypes,
+    GasLimit: Unwrap<Output = u64>,
+    TransferredValue: Unwrap<Output = E::Balance>,
+{
+    /// Finalizes the call builder to call a function.
+    pub fn params(self) -> CallParams<E, Args, RetType> {
+        CallParams {
+            callee: self.callee.value(),
+            gas_limit: self.gas_limit.unwrap_or_else(|| 0),
+            transferred_value: self
+                .transferred_value
+                .unwrap_or_else(|| E::Balance::from(0)),
+            return_type: Default::default(),
```

Comment:
Set the return type?

Reply:
again the return type field is only used to compile time signature derivation. Not really a thing anymore during runtime.